File: my_reports/my_reports/report/cost_center_wise/cost_center_wise.js
// Copyright (c) 2016, h@ci and contributors
// For license information, please see license.txt

frappe.query_reports["Cost Center Wise"] = {
	"filters": [
		{
			fieldname: "sales_person",
			label: __("Sales Person"),
			fieldtype: "Link",
			options: "Sales Person"
		},
		{
			fieldname: "doc_type",
			label: __("Document Type"),
			fieldtype: "Select",
			options: "Sales Order\nDelivery Note\nSales Invoice",
			default: "Sales Order"
		},
		{
			fieldname: "from_date",
			label: __("From Date"),
			fieldtype: "Date",
			default: frappe.defaults.get_user_default("year_start_date"),
		},
		{
			fieldname:"to_date",
			label: __("To Date"),
			fieldtype: "Date",
			default: get_today()
		},
		{
			fieldname:"cost_center",
			label: __("Cost Center"),
			fieldtype: "Link",
			options: "Cost Center",
		},
		{
			fieldname:"brand",
			label: __("Brand"),
			fieldtype: "Link",
			options: "Brand",
		},
		{
			fieldname:"customer",
			label: __("Customer"),
			fieldtype: "Link",
			options: "Customer",
		},
	]
}

File: my_reports/my_reports/report/cost_center_wise/cost_center_wise.py
# Copyright (c) 2013, h@ci and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe	
from frappe import _

def execute(filters=None):
	columns = [
		_("Account") + ":Link/Account:140",
		_("Posting Date") + ":Date:100",
		_("Credit") + ":Currency:100",
		_("Debit") + ":Currency:100",
		_("Remarks") + "::400",]
	data = frappe.db.sql("""
		select
		    account, posting_date, credit, debit, remarks
		from
		    `tabGL Entry`
		where
		    cost_center IS NOT NULL AND posting_date>"%s" AND posting_date<"%s"
			""" % (filters["from_date"], filters["to_date"]))
	
	return columns, data